pymatgen/io/vasp/inputs.py

```diff
@@ -362,55 +376,57 @@ def from_str(cls, data, default_names=None, read_velocities=True) -> Self:
 
             for i_line_symbols in range(6, 5 + n_lines_symbols):
                 symbols.extend(lines[i_line_symbols].split())
+
             n_atoms = []
             iline_natoms_start = 5 + n_lines_symbols
             for iline_natoms in range(iline_natoms_start, iline_natoms_start + n_lines_symbols):
                 n_atoms.extend([int(i) for i in lines[iline_natoms].split()])
 
-            for i, nat in enumerate(n_atoms):
-                atomic_symbols.extend([symbols[i]] * nat)
+            for idx, n_atom in enumerate(n_atoms):
+                atomic_symbols.extend([symbols[idx]] * n_atom)
+
             ipos = 5 + 2 * n_lines_symbols
 
-        pos_type = lines[ipos].split()[0]
+        pos_type: str = lines[ipos].split()[0]
 
-        has_selective_dynamics = False
+        has_selective_dynamics: bool = False
         # Selective dynamics
         if pos_type[0] in "sS":
             has_selective_dynamics = True
             ipos += 1
             pos_type = lines[ipos].split()[0]
 
-        cart = pos_type[0] in "cCkK"
-        n_sites = sum(n_atoms)
+        cart: bool = pos_type[0] in "cCkK"
+        n_sites: int = sum(n_atoms)
 
         # If default_names is specified (usually coming from a POTCAR), use
         # them. This is in line with VASP's parsing order that the POTCAR
         # specified is the default used.
-        if default_names:
+        if default_names is not None:
```

Comment:
The default value of `default_names` seems inconsistent in: https://github.com/materialsproject/pymatgen/blob/7f96ef2c6c4e753a5dafbac4fde6f5a136c0aec0/pymatgen/core/structure.py#L2860-L2863

Changed to `None` in 1a1752a38a526f39b2c12279f9f15fbad593a73d.